checks.bitrise.yml

```diff
@@ -0,0 +1,57 @@
+format_version: "11"
+default_step_lib_source: https://github.com/bitrise-io/bitrise-steplib.git
+
+workflows:
+  lint:
+    steps:
+    - change-workdir@1:
+        inputs:
+        - path: $STEP_DIR
+    - script:
+        title: YAML lint
+        inputs:
+        - content: |-
+            #!/bin/env bash
+            set -ex
+            pip3 install yamllint
+            yamllint --format colored . # Config file is implicitly set via $YAMLLINT_CONFIG_FILE
+    - script@1:
+        title: Audit step
+        run_if: '{{enveq "SKIP_STEP_YML_VALIDATION" "false"}}'
+        inputs:
+        - content: |-
+            #!/bin/env bash
+            set -ex
+            pwd
+
+            stepman audit --step-yml ./step.yml
+    - git::https://github.com/bitrise-steplib/steps-validate-json-schema.git@main:
+        title: Validate JSON schema
+        inputs:
+        - schema_url: https://raw.githubusercontent.com/bitrise-io/bitrise-json-schemas/main/step.schema.json
+        - yaml_path: ./step.yml
+        - warning_patterns: |-
+            I\[#\] S\[#/additionalProperties\] additionalProperties .+ not allowed
+            I\[#\] S\[#/required\] missing properties: .+
+            I\[#/summary\] S\[#/properties/summary/pattern\] does not match pattern "\^\.\{1,100\}\$"
+            I\[#/deps/(brew|apt_get)+/\d+/(name|bin_name)+\] S\[#/definitions/(BrewDepModel|AptGetDepModel)+/properties/(name|bin_name)+/not\] not failed
+            I\[#/(inputs|outputs)+/\d+/opts\] S\[#/definitions/EnvVarOpts/required\] missing properties: "summary"
+            I\[#/(inputs|outputs)+/\d+/opts/summary\] S\[#/definitions/EnvVarOpts/properties/summary/minLength\] length must be >= 1, but got 0
+            I\[#/inputs/\d+/.+\] S\[#/definitions/InputEnvVar/additionalProperties/type\] expected .+, but got .+
+            I\[#/inputs/\d+/opts/value_options\] S\[#/definitions/EnvVarOpts/properties/value_options/minItems\] minimum 2 items allowed, but found \d+ items
```

Comment:
I have a small question regarding this. It looks good as it is but I just want to check something. 

The `validate-json-schema` is dumb meaning that the rules are need to be fed into it. Will this step only be used by the unified CI? I am wondering if you already see a use case where we need to copy paste these rules to another place too. Because if you do then it might be better to put these somewhere else. If not then everything is good. 😄

Reply:
These are patterns we use to mark certain issues as warnings.

Ideally, this list should be empty so that we treat every JSON schema issue as an error.
I did not want to update all the unified ci steps to fix all the current issues, this way we 'omit' these issues.
I created a separate ticket to fix all the current issues with the unified ci steps and remove these warning patterns.

If we integrate this check into the StepLib CI we can reuse these patterns to mark already existing issues as warnings too.